src/command/process_update_task.rs

```diff
@@ -134,23 +184,109 @@ impl UpdateProcessor {
 
                 self.find_city_message().await?;
 
-                Ok(())
+                Ok(None)
+            }
+            Command::Start => {
+                self.start_message().await?;
+                Ok(None)
+            }
+            Command::CurrentDefaultCity => {
+                let text = match self.chat.default_city_id {
+                    Some(id) => match self.repo.search_city_by_id(&id).await {
+                        Ok(city) => format!("Your default city is {}", city),
+                        Err(_) => "You do not have default city".to_string(),
+                    },
+                    None => "You do not have default city".to_string(),
+                };
+                self.send_message(&text).await?;
+
+                Ok(None)
+            }
+            Command::SetDefaultCity => {
+                self.set_city().await?;
+                Ok(None)
             }
-            Command::Start => self.start_message().await,
-            Command::SetDefaultCity => self.set_city().await,
             Command::Default => match self.chat.default_city_id {
                 Some(id) => {
                     let city = self.repo.search_city_by_id(&id).await?;
 
-                    self.get_weather(city).await
+                    self.get_weather(city).await?;
+
+                    Ok(None)
                 }
                 None => {
                     self.set_city().await?;
 
-                    self.not_default_message().await
+                    self.not_default_message().await?;
+
+                    Ok(None)
                 }
             },
-            _ => self.unknown_command().await,
+            Command::Schedule => {
+                self.schedule_weather().await?;
+                Ok(None)
+            }
+            Command::SetOffset => {
+                self.set_offset().await?;
+                Ok(None)
+            }
+            Command::UnSchedule => self.unschedule().await,
+            _ => {
+                self.unknown_command().await?;
+                Ok(None)
+            }
+        }
+    }
+
+    async fn unschedule(&self) -> Result<Option<Vec<Forecast>>, BotError> {
```

Comment:
maybe it can be improved in the future. but currently a user can unschedule only all forecasts

Reply:
https://github.com/pxp9/weather_bot_rust/issues/31